contracts/ProcessDefinition.sol

```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier: MIT
+// compiler version must be greater than or equal to 0.8.20 and less than 0.9.0
+pragma solidity ^0.8.20;
+
+
+contract ProcessDefinition {
+    address public factory;
+    // Mapping of started process instance IDs to their statuses.
+    mapping(string => ProcessInstanceStatus) public startedProcessInstances;
+    // Mapping of started process instance IDs to their owners.
+    mapping(string => address) public processInstanceOwners;
+    // List of all started process instance IDs.
+    string[] private list;
+
+    address public owner;
+    uint public processPrice;
+
+    address public feeClaimer;
+    // 5% fee
+    uint public fee = 5;
+
+    enum ProcessInstanceStatus {
+        Created,
+        InProgress,
+        Completed,
+        Failed
+    }
+
+    constructor() {
+        factory = msg.sender;
+    }
+
+    event ProcessInstanceStarted(string processInstanceId);
+    event ProcessInstanceCompleted(string processInstanceId);
+    event ProcessInstanceFailed(string processInstanceId);
+    event TransferOwnership(address newOwner);
+    event ChangeFeeClaimer(address newFeeClaimer);
+    event Deposit(address from, uint amount);
+    event Withdraw(address to, uint amount);
+    event Transfer(address to, uint amount);
+
+
+    modifier onlyOwner() {
+        require(msg.sender == owner, "Only owner can call this function.");
+        _;
+    }
+
+    // Set owner of ProcessDefinition. Can be called by factory only.
+
+    function initialize(address _owner, address _feeClaimer, uint _price) external {
+        require(msg.sender == factory, "Only factory can set onwer");
+        require(owner == address(0), "Contract already has owner");
+        owner = _owner;
+        feeClaimer = _feeClaimer;
+        processPrice = _price;
+    }
+
+    function changeFeeClaimer(address _feeClaimer) external {
+        require(msg.sender == feeClaimer, "Only feeClaimer can change feeClaimer");
+        feeClaimer = _feeClaimer;
+        emit ChangeFeeClaimer(_feeClaimer);
+    }
+
+    function transferOwnership(address _newOwner) external onlyOwner {
+        owner = _newOwner;
+        emit TransferOwnership(_newOwner);
+    }
+
+    // Starts a new process instance.
+    function start(string memory processInstanceId) external payable {
```

Comment:
please create end method, which would substract fee + process price from locked tokens.

Reply:
fee transfer happens on the start event (see below). It's not as optimal for gas, but it's easier to handle. As for the withdraw for the process owner, we have an "withdraw" function which he can call at any time and get his tokens with the fee deducted.

So do we need the "end" function?